core/lib/config/src/configs/en_config.rs

```diff
@@ -0,0 +1,40 @@
+use std::num::{NonZeroU32, NonZeroU64, NonZeroUsize};
+
+use serde::Deserialize;
+use zksync_basic_types::{
+    commitment::L1BatchCommitmentMode, url::SensitiveUrl, L1ChainId, L2ChainId,
+};
+
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+pub struct Pruning {
+    pub enabled: bool,
+    pub chunk_size: Option<u32>,
+    pub removal_delay_sec: Option<NonZeroU64>,
+    pub data_retention_sec: Option<u64>,
+}
+
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+pub struct SnapshotRecovery {
+    pub enabled: bool,
+    pub postgres_max_concurrency: Option<NonZeroUsize>,
+    pub tree_chunk_size: Option<u64>,
+}
+
+/// Temporary config for initializing external node, will be completely replaced by consensus config later
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+pub struct ENConfig {
+    // Genesis
+    pub l2_chain_id: L2ChainId,
+    pub l1_chain_id: L1ChainId,
+    pub l1_batch_commit_data_generator_mode: L1BatchCommitmentMode,
+
+    // Main node configuration
+    pub main_node_url: SensitiveUrl,
+    pub main_node_rate_limit_rps: Option<NonZeroUsize>,
+
+    // En specific components
+    pub commitment_generator_max_parallelism: Option<NonZeroU32>,
+    pub tree_api_remote_url: Option<String>,
+    pub snapshot_recovery: Option<SnapshotRecovery>,
+    pub pruning: Option<Pruning>,
```

Comment:
Since `SnapshotRecovery` and `Pruning` both have default values, it could make sense making these fields non-optional and use `#[serde(default)]` / corresponding glue for Protobuf. There's a similar problem with [`Secrets`](https://github.com/matter-labs/zksync-era/blob/49198f695a93d24a5e2d37a24b2c5e1b6c70b9c5/core/lib/config/src/configs/secrets.rs#L19) and `DatabaseSecrets`.

Reply:
it makes it slightly harder to work with protobuf configs.
Probably we can address it in the next step